### DISCUSSION TITLE
Feature request: Add old photo restoration

### DISCUSSION THREAD

**Discussion by Darin755** [upscayl/upscayl#470] [Ideas]
I have a bunch of old black and white photos of family members that were taken on film.  

When I run them though upscayl they just get mutilated as it apparently doesn't know what to do with natural defects. It would be nice to be able upscayl the photos into color to help me and others see into the past.

**Comment by NayamAmarshe** [comment]
That's actually something we can MOST DEFINITELY consider for Upscayl Cloud :)

Upscayl Desktop however, is a different beast altogether and it's a bit less flexible when it comes to something like this because we're limited by hardware and also the platforms and everything.

**Reply by Darin755** [reply]
Well I personally will never use "upscayl cloud" as its a good example of SaSS (service as a software substitute).

**Reply by etlweather** [reply]
Same here, most usecases I have where that would be useful, we would want to run on our own equipment.

**Reply by aaronliu0130** [reply]
Well, we probably shouldn't increase the binaries we ship either. Maybe we should add a mode buried in settings to allow the user to specify custom commands (and arguments) and run that string + the file name selected to Upscayl (with only normal upscayl and no model selection step)? And perhaps add configuration guides in the wiki?

**Reply by etlweather** [reply]
> Well, we probably shouldn't increase the binaries we ship either.

Or make it a different app like *fixaly*. 

Personally, a different app wouldn't be a problem. Having a desktop, no cloud, usable photo restoration app would be great. I used, for example, [Sanster / lama-cleaner](https://github.com/Sanster/lama-cleaner) and it was useful. It worked for me because I could run it in a Docker container but that's not really end-user friendly.

Having an app that is as simple to install as Upscayl for photo restoration would be great.

Personally, I don't see this as a must to be part of Upscayl.

**Reply by aaronliu0130** [reply]
Sure. Still, I think adding custom commands would be a great alternative to Upscayl Cloud for service dislikers.

**Reply by Darin755** [reply]
I think it would be a good idea to setup some sort of plugin support. You could have a base use case out of the box and then have addons you can enable to add functionality.
